File: app/src/main/java/com/marked/vifo/fragment/signup/SignUpEmailFragment.java
package com.marked.vifo.fragment.signup;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.util.Patterns;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.marked.vifo.R;
import com.marked.vifo.helper.Utils;
import com.rengwuxian.materialedittext.MaterialEditText;

import java.io.UnsupportedEncodingException;
import java.net.URLEncoder;

/**
 * A simple {@link Fragment} subclass.
 * Activities that contain this fragment must implement the
 * {@link SignUpEmailFragmentInteraction} interface
 * to handle interaction events.
 * Use the {@link SignUpEmailFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class SignUpEmailFragment extends Fragment {
    private SignUpEmailFragmentInteraction mListener;
    private MaterialEditText mEmail;

    public SignUpEmailFragment() {
        // Required empty public constructor
    }

    public static SignUpEmailFragment newInstance() {
        return new SignUpEmailFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_sign_up_email, container, false);
        mEmail = (MaterialEditText) rootView.findViewById(R.id.emailEditText);
        rootView.findViewById(R.id.nextButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (!Utils.isOnline(getActivity())){
                    Utils.showNoConnectionDialog(getActivity());
                } else {
                    String email = mEmail.getText().toString().trim();
                    if (Utils.isEmpty(email))
                        Toast.makeText(getActivity(), "Please enter an email", Toast.LENGTH_LONG).show();
                    else if (Patterns.EMAIL_ADDRESS.matcher(email).matches())
                        onNextPressed(email);
                    else
                        Toast.makeText(getActivity(), "Invalid email adress", Toast.LENGTH_LONG).show();
                }
            }
        });


        // Inflate the layout for this fragment
        return rootView;
    }

    // TODO: Rename method, update argument and hook method into UI event
    public void onNextPressed(String email) {
        if (mListener != null) {
            try {
                mListener.onSaveEmail(URLEncoder.encode(email, "UTF-8"));
            } catch (UnsupportedEncodingException e) {
                e.printStackTrace();
            }
        }
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof SignUpEmailFragmentInteraction) {
            mListener = (SignUpEmailFragmentInteraction) context;
        } else {
            throw new RuntimeException(
                    context.toString() + " must implement SignUpEmailFragmentInteraction");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    public interface SignUpEmailFragmentInteraction {
        // TODO: Update argument type and name
        void onSaveEmail(String email);
    }
}

File: app/src/main/java/com/marked/vifo/ui/fragment/signup/SignUpEmailFragment.kt
package com.marked.vifo.ui.fragment.signup

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.Patterns
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.marked.vifo.R
import com.marked.vifo.helper.Toast
import com.marked.vifo.helper.Utils
import kotlinx.android.synthetic.main.fragment_sign_up_email.*
import java.io.UnsupportedEncodingException
import java.net.URLEncoder

/**
 * A simple [Fragment] subclass.
 * Activities that contain this fragment must implement the
 * [SignUpEmailFragmentInteraction] interface
 * to handle interaction events.
 * Use the [SignUpEmailFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class SignUpEmailFragment : Fragment() {
	private var mListener: SignUpEmailFragmentInteraction? = null

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
		val rootView = inflater.inflate(R.layout.fragment_sign_up_email, container, false)
		hiUser.text = "Beautiful name ${arguments.getString(NAME_TAG)},"
		nextButton.setOnClickListener {
			if (!Utils.isOnline(activity)) {
				Utils.showNoConnectionDialog(activity)
			} else {
				val email = emailEditText.text.toString().trim { it <= ' ' }
				if (Utils.isEmpty(email))
					activity.Toast("Please enter an email")
				else if (Patterns.EMAIL_ADDRESS.matcher(email).matches())
					onNextPressed(email)
				else
					activity.Toast("Invalid email adress")
			}
		}
		// Inflate the layout for this fragment
		return rootView
	}

	// TODO: Rename method, update argument and hook method into UI event
	fun onNextPressed(email: String) {
		if (mListener != null) {
			try {
				mListener!!.onSaveEmail(URLEncoder.encode(email, "UTF-8"))
			} catch (e: UnsupportedEncodingException) {
				e.printStackTrace()
			}

		}
	}

	override fun onAttach(context: Context?) {
		super.onAttach(context)
		if (context is SignUpEmailFragmentInteraction)
			mListener = context as SignUpEmailFragmentInteraction?
		else
			throw RuntimeException(context!!.toString() + " must implement SignUpEmailFragmentInteraction")
	}

	override fun onDetach() {
		super.onDetach()
		mListener = null
	}

	interface SignUpEmailFragmentInteraction {
		// TODO: Update argument type and name
		fun onSaveEmail(email: String)
	}

	companion object {
		val NAME_TAG = "NAME"
		fun newInstance(name: String): SignUpEmailFragment {
			val bundle = Bundle()
			bundle.putString(NAME_TAG, name)
			val fragment = SignUpEmailFragment()
			fragment.arguments = bundle
			return fragment
		}
	}
}// Required empty public constructor
